source/auth.js

```diff
@@ -1,13 +1,80 @@
 "use strict";
 
-function generateBasicAuthHeader(username, password) {
+const crypto = require("crypto");
+
+let username, password;
```

Comment:
Not sure I love the idea of these being outside of function scope..

Reply:
See https://github.com/perry-mitchell/webdav-client/pull/152/commits/8c52e4ffd8d7265abcc4f01c6e956ef861cb80c8#diff-1995eac40392e12643bbf44abad3cf8c